src/Core/Compiler/NodeConverter.php

```diff
@@ -107,7 +107,12 @@ public function convert(NodeInterface $node)
     protected function convertEscapingNode(EscapingNode $node)
     {
         $configuration = $this->convert($node->getNode());
-        $configuration['execution'] = sprintf('htmlspecialchars(%s, ENT_QUOTES)', $configuration['execution']);
+        $configuration['execution'] = sprintf(
+            'call_user_func_array( function ($var) { ' .
+            'return (is_string($var) || (is_object($var) && method_exists($var, \'__toString\')) ' .
```

Comment:
I'm not sure I agree - could you illustrate a case where a non-string value would potentially result in unsafe output if (implicitly) cast to a string w/o doing `htmlspecialchars()`?